compiler/rustc_parse/src/lexer/unescape_error_reporting.rs

```diff
@@ -29,9 +31,9 @@ pub(crate) fn emit_unescape_error(
     let span = {
         let Range { start, end } = range;
         let (start, end) = (start as u32, end as u32);
-        let lo = span_with_quotes.lo() + BytePos(start + 1);
+        let lo = span.lo() + BytePos(start);
         let hi = lo + BytePos(end - start);
-        span_with_quotes.with_lo(lo).with_hi(hi)
+        span.with_lo(lo).with_hi(hi)
```

Comment:
This code is still doing arithmetic on arbitrary spans.
`let span = self.mk_sp(content_start, content_end)` will turn `span` into `override_span` in exactly the same way as `span_with_quotes`.

All the range arithmetic should be done on `content_start`/`content_end` which are guaranteed to be from lexer, and `mk_sp` should only be called in the end on its results (and it may drop those results and replace them with `override_span`).

Reply:
Moved this logic to the caller to operate on `content_*` directly.